src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/CertificateRequest.Load.cs

```diff
@@ -0,0 +1,393 @@
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    public sealed partial class CertificateRequest
+    {
+        private const CertificateRequestLoadOptions AllOptions =
+            CertificateRequestLoadOptions.SkipSignatureValidation |
+            CertificateRequestLoadOptions.UnsafeLoadCertificateExtensions;
+
+        public static unsafe CertificateRequest LoadSigningRequestPem(
+            string pkcs10Pem,
+            HashAlgorithmName signerHashAlgorithm,
+            CertificateRequestLoadOptions options = CertificateRequestLoadOptions.Default,
+            RSASignaturePadding? signerSignaturePadding = null)
+        {
+            ArgumentNullException.ThrowIfNull(pkcs10Pem);
+
+            return LoadSigningRequestPem(
+                pkcs10Pem.AsSpan(),
+                signerHashAlgorithm,
+                options,
+                signerSignaturePadding);
+        }
+
+        public static unsafe CertificateRequest LoadSigningRequestPem(
+            ReadOnlySpan<char> pkcs10Pem,
+            HashAlgorithmName signerHashAlgorithm,
+            CertificateRequestLoadOptions options = CertificateRequestLoadOptions.Default,
+            RSASignaturePadding? signerSignaturePadding = null)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(signerHashAlgorithm.Name, nameof(signerHashAlgorithm));
+
+            if ((options & ~AllOptions) != 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(options), options, SR.Argument_InvalidFlag);
+            }
+
+            foreach ((ReadOnlySpan<char> contents, PemFields fields) in new PemEnumerator(pkcs10Pem))
+            {
+                if (contents[fields.Label].SequenceEqual(PemLabels.Pkcs10CertificateRequest))
+                {
+                    byte[] rented = ArrayPool<byte>.Shared.Rent(fields.DecodedDataLength);
+
+                    if (!Convert.TryFromBase64Chars(contents[fields.Base64Data], rented, out int bytesWritten))
+                    {
+                        Debug.Fail("Base64Decode failed, but PemEncoding said it was legal");
+                        throw new UnreachableException();
+                    }
+
+                    try
+                    {
+                        return LoadSigningRequest(
+                            rented.AsSpan(0, bytesWritten),
+                            permitTrailingData: false,
+                            signerHashAlgorithm,
+                            out _,
+                            options,
+                            signerSignaturePadding);
+                    }
+                    finally
+                    {
+                        ArrayPool<byte>.Shared.Return(rented);
+                    }
+                }
+            }
+
+            throw new CryptographicException(SR.Cryptography_NoPemOfLabel, PemLabels.Pkcs10CertificateRequest);
+        }
+
+        public static unsafe CertificateRequest LoadSigningRequest(
```

Comment:
I don't see anything unsafe here.

```suggestion
        public static CertificateRequest LoadSigningRequest(
```